### ISSUE TITLE
Wrong function signature in Ghidra decompiler

### ISSUE THREAD

**Issue by cookpoo78** [radareorg/r2ghidra-dec#357]
Hey guys, why does the decompiler interpret the function arguments wrong? the disassembler does it well though and comment the right parameters. 
It's even not possible to override the function signature to make it right (for these cases where the interpretation is wrong or incomplete)..
Needless to say, it's just [VirtualAlloc](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualalloc) with a known signature.
So to sum up:
1. If the disassembler gets it right, why doesn't the decompiler (that I guess anyway relies on the disassembling output)? 
2. Why can't I modify the function signature in the decompiler view?

![image](https://github.com/user-attachments/assets/9637e08e-9514-425f-bc21-24ba02ba56fd)
